### PR TITLE
neon_local: Use clap derive macros to parse the CLI args

### PR DESCRIPTION
**Pull request by hlinnaka**
This is easier to work with.

### CODE REVIEW

File: control_plane/src/bin/neon_local.rs [hunk 1]
```diff
@@ -56,10 +56,640 @@ const DEFAULT_PAGESERVER_ID: NodeId = NodeId(1);
...
+#[derive(clap::Args)]
+#[clap(about = "Stop an endpoint")]
+struct EndpointStopCmdArgs {
+    #[clap(help = "Postgres endpoint id")]
+    endpoint_id: String,
+
+    //.help("Also delete data directory (now optional, should be default in future)")
+    //                      .long("destroy")
+    //                      .action(ArgAction::SetTrue)
+    //                    .required(false)
```
[the remark this answers is not in the document]
Comment (hlinnaka) [reply control_plane/src/bin/neon_local.rs]: fixed, thanks

File: control_plane/src/bin/neon_local.rs
[the remark this answers is not in the document]
Comment (hlinnaka) [reply control_plane/src/bin/neon_local.rs:1]: Yeah, that would make sense. I'm going to leave that for a follow up PR, it doesn't need to be done as part of this one.

File: control_plane/src/bin/neon_local.rs [hunk 2]
```diff
@@ -56,10 +56,632 @@ const DEFAULT_PAGESERVER_ID: NodeId = NodeId(1);
...
+    #[arg(value_enum)]
+    #[clap(long, default_value_t = StopMode::Fast)]
+    mode: StopMode,
+}
+
+//        .short('m')
+//        .value_parser(["fast", "immediate"])
+//        .default_value("fast")
+//        .help("If 'immediate', don't flush repository data at shutdown")
+//        .required(false)
```
Comment (yliang412) [inline control_plane/src/bin/neon_local.rs:131]:
```suggestion
//        .short('m')
//        .value_parser(["fast", "immediate"])
//        .default_value("fast")
//        .help("If 'immediate', don't flush repository data at shutdown")
//        .required(false)
```

**Comment by tristan957** [comment]
It would be awesome to get this done for compute_ctl as well.

**Review by yliang412 — APPROVED** [review]
Looks good!

File: control_plane/src/bin/neon_local.rs (hunk 2 above)
Comment (hlinnaka) [reply control_plane/src/bin/neon_local.rs:131]: no-op suggestion?

Comment (yliang412) [reply control_plane/src/bin/neon_local.rs:131]: Fixed 🤣

Comment (hlinnaka) [reply control_plane/src/bin/neon_local.rs:131]: ah, thanks!

**Pull request merged**